src/System.Management.Automation/namespaces/FunctionProvider.cs

```diff
@@ -41,6 +41,29 @@ public FunctionProvider()
 
         #endregion Constructor
 
+        #region CmdletProvider overrides
+
+        /// <summary>
+        /// Starts the Function provider. It sets the PathSeparator property.
+        /// </summary>
+        /// <param name="providerInfo">
+        /// The ProviderInfo object that holds the provider's configuration.
+        /// </param>
+        /// <returns>
+        /// The updated ProviderInfo object that holds the provider's configuration.
+        /// </returns>
+        protected override ProviderInfo Start(ProviderInfo providerInfo)
```

Comment:
Similar method should be added for `Environment`, `Alias`, `Certificate` and `WSMan` providers.

Reply:
I have yet to add it for the `WSMan` provider. The reason is that `WSMan` provider is in a different namespace, thus would require to change classes in the [Utils file](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/Utils.cs) to have public scope

Reply:
What you should be the outcome here? Should I change the relevant classes' scope?

Reply:
Good point. I think we can use a new instance of `ReadOnlyCollection<string>` for `WSMan`.